### PR TITLE
Modify VmCore::step to return a PyErr

### PR DESCRIPTION
**Pull request by pefontana**
* Modify VmCore::step to return a PyErr instead of a VirtualMachineError
* Remove unused fn Utils::to_vm_error

### CODE REVIEW

**Review by fmoletta — APPROVED** [review]
Great Work!

File: src/ids.rs [hunk 1]
```diff
@@ -1127,13 +1119,11 @@ memory[fp] = ids.ok_ref
...
-                py_result.map_err(|err| to_vm_error(err, py)),
-                Err(to_vm_error(
-                    PyValueError::new_err(VirtualMachineError::NoRegisterInReference.to_string()),
-                    py
-                ))
-            );
+            assert!(py_result.unwrap_err().to_string().contains(
+                &VirtualMachineError::NoRegisterInReference
+                    .to_string()
+                    .to_string()
```
Comment (Oppen) [inline src/ids.rs:1125]: Why do you need to `.to_string()` twice?

**Review by Oppen — COMMENTED** [review]
Very nice! I have a minor question as part of this review. Additionally, I suggest rewriting `to_py_error` as an implementation of the `From` trait for `PyErr` and `VirtualMachineError` to reduce verbosity (instead of a myriad of `.map_err(to_py_error)` we would simply have `.into()`).

File: src/ids.rs (hunk 1 above)
Comment (pefontana) [reply src/ids.rs]: Just a copy paste error.
Removed [a0fe931](https://github.com/lambdaclass/cairo-rs-py/pull/175/commits/a0fe93135a090b3f0e7351f0ba4d1bf5150a19eb)

Comment (pefontana) [reply src/ids.rs]: https://github.com/lambdaclass/cairo-rs-py/issues/180

**Review by Oppen — APPROVED** [review]

**Pull request merged**